ValveResourceFormat/Serialization/VfxEval.cs

```diff
@@ -0,0 +1,503 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using ValveResourceFormat.ThirdParty;
+
+namespace ValveResourceFormat.Serialization.VfxEval
+{
+    public class VfxEval
+    {
+        // parsed data assigned here
+        public string DynamicExpressionResult { get; private set; }
+        // parse the input one line at a time
+        private readonly List<string> DynamicExpressionList = new();
+
+        // check externally if we have problems parsing
+        public bool ErrorWhileParsing { get; private set; }
+        public string ErrorMessage { get; private set; }
+
+
+        // function reference, name and number of arguments
+        private readonly (string, int)[] FUNCTION_REF = {
+            ("sin",        1),     // 00
+            ("cos",        1),     // 01
+            ("tan",        1),     // 02
+            ("frac",       1),     // 03
+            ("floor",      1),     // 04
+            ("ceil",       1),     // 05
+            ("saturate",   1),     // 06
+            ("clamp",      3),     // 07
+            ("lerp",       3),     // 08
+            ("dot4",       2),     // 09
+            ("dot3",       2),     // 0A
+            ("dot2",       2),     // 0B
+            ("log",        1),     // 0C
+            ("log2",       1),     // 0D
+            ("log10",      1),     // 0E
+            ("exp",        1),     // 0F
+            ("exp2",       1),     // 10
+            ("sqrt",       1),     // 11
+            ("rsqrt",      1),     // 12
+            ("sign",       1),     // 13
+            ("abs",        1),     // 14
+            ("pow",        2),     // 15
+            ("step",       2),     // 16
+            ("smoothstep", 3),     // 17
+            ("float4",     4),     // 18
+            ("float3",     3),     // 19
+            ("float2",     2),     // 1A
+            ("time",       0),     // 1B
+            ("min",        2),     // 1C
+            ("max",        2),     // 1D
+            ("SrgbLinearToGamma",1), // 1E
+            ("SrgbGammaToLinear",1), // 1F
+            ("random",     2),     // 20
+            ("normalize",  1),     // 21
+            ("length",     1),     // 22
+            ("sqr",        1),     // 23
+            ("TextureSize",1),     // 24
+        };
```

Comment:
There's 3 more in dota: rotation2d, rotate2d, sincos

Reply:
I found them I'll write them in